### ISSUE TITLE
require-returns-check on non-void type should give error even if 1 valid return

### ISSUE THREAD

**Issue by kkmuffme** [gajus/eslint-plugin-jsdoc#893]
if type is non-void but void return is used, we should get an error.

When we do this:

```js
/**
 * @param {string} message
 * @return {Boolean}
 */
function abc( message ) {
    if ( message ) {
        return;
    }
}
```

We already get an error that the "return" is missing.

However when I do this, I expect to also get an error for the "return;"

```js
/**
 * @param {string} message
 * @return {Boolean}
 */
function abc( message ) {
    if ( message ) {
        return;
    }
    
    return false;
}
```

**Comment by brettz9** [comment]
The related issue #892 was fixed.

As to why this is still an issue...

It has been the approach of `eslint-plugin-jsdoc` to only check preferred tag names, e.g., `@returns` or `@return`. This works if setting `@return` to `@returns` in your example—or keeping it as is and adding this for your ESLint config settings:

```js
      settings: {
        jsdoc: {
          tagNamePreference: {
            returns: 'return',
          },
        },
      },
```

I am thus closing though feel free to comment further as needed.
